### PR TITLE
Added unit test coverage for SignificantTerms

### PR DESCRIPTION
**Pull request by markharwood**
Covers GlobalOrdinalsSignificantTermsAggregator, GlobalOrdinalsSignificantTermsAggregator.WithHash, SignificantLongTermsAggregator and SignificantStringTermsAggregator

Removed integration test

Relates #22278

### CODE REVIEW

**Review by colings86 — COMMENTED** [review]
I think we should add some of the tests from the removed integration test suite. Specifically I think we should test with include/exclude, with a different heuristic and unmapped/partially unmapped fields

**Comment by rjernst** [comment]
@markharwood Do you want to address Colin's feedback and get this in?

**Comment by markharwood** [comment]
Looking at this now

**Comment by markharwood** [comment]
@colings86 Care to take another look?

File: core/src/test/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregatorTests.java [hunk 1]
```diff
@@ -237,6 +228,60 @@ public void testNumericSignificance() throws IOException {
...
+            SignificantTermsAggregationBuilder sigAgg = new SignificantTermsAggregationBuilder("sig_text", null).field("unmapped_field");
+            sigAgg.executionHint(randomExecutionHint());
+
+            try (IndexReader reader = DirectoryReader.open(w)) {
+                assertEquals("test expects a single segment", 1, reader.leaves().size());
+                IndexSearcher searcher = new IndexSearcher(reader);
+
+                // Search "odd"
+                SignificantTerms terms = searchAndReduce(searcher, new TermQuery(new Term("text", "odd")), sigAgg, textFieldType);
+
```
Comment (colings86) [inline core/src/test/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregatorTests.java]: should we check that the number of buckets returned is zero instead to ensure nothing weird is going on?

**Review by colings86 — APPROVED** [review]
Left one comment but LGTM

**Comment by markharwood** [comment]
jenkins test this

**Pull request merged**